x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformStats.java

```diff
@@ -233,4 +237,78 @@ public DataFrameTransformCheckpointingInfo getCheckpointingInfo() {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public enum State implements Writeable {
+
+        STARTED, INDEXING, ABORTING, STOPPING, STOPPED, FAILED;
+
+        public static State fromString(String name) {
+            return valueOf(name.trim().toUpperCase(Locale.ROOT));
+        }
+
+        public static State fromStream(StreamInput in) throws IOException {
+            return in.readEnum(State.class);
+        }
+
+        public static State fromComponents(DataFrameTransformTaskState taskState, IndexerState indexerState) {
+
+            if (taskState == null || taskState == DataFrameTransformTaskState.STOPPED) {
+                return STOPPED;
+            } else if (taskState == DataFrameTransformTaskState.FAILED) {
+                return FAILED;
+            } else {
+
+                // Note: indexer state CAN be null if the task state is stopped or failed
+                assert(indexerState != null);
```

Comment:
nit: the note and the assert look contradicting to me, indexerState should not be null, if there is no indexer it should be `STOPPED`

Reply:
I don't think they were contradicting, as the "if the task state is stopped or failed" bit refers to the _other_ code branches.

But since the way it was phrased was confusing I changed it in 3f35a2b6c0a3dc98ee03a3ffe6845adbef118c92 to only refer to what's expected at the comment location.